aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/Translator.java

```diff
@@ -111,35 +113,22 @@ static UntagLogGroupRequest translateToUntagLogGroupRequest(final String logGrou
                 .build();
     }
 
-    static ResourceModel translateForRead(final DescribeLogGroupsResponse response, final ListTagsLogGroupResponse tagsResponse) {
-        final String logGroupName = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::logGroupName)
-                .filter(Objects::nonNull)
-                .findAny()
-                .orElse(null);
-        final String logGroupArn = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::arn)
-                .filter(Objects::nonNull)
-                .findAny()
-                .orElse(null);
-        final Integer retentionInDays = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::retentionInDays)
-                .filter(Objects::nonNull)
-                .findAny()
-                .orElse(null);
-        final String kmsKeyId = streamOfOrEmpty(response.logGroups())
-                .map(software.amazon.awssdk.services.cloudwatchlogs.model.LogGroup::kmsKeyId)
-                .filter(Objects::nonNull)
+    static ResourceModel translateForRead(final DescribeLogGroupsResponse response, final ListTagsLogGroupResponse tagsResponse, final String requestLogGroupName) {
+        LogGroup matchedLogGroup = streamOfOrEmpty(response.logGroups())
+                .filter(logGroup -> logGroup.logGroupName() != null && logGroup.logGroupName().equals(requestLogGroupName))
```

Comment:
lets break this out into two filters. .filter(not null).filter(name == name). Also, what is the point of the null check here? when would a log group name be null?

Reply:
the test `testTranslateForReadResponse_LogGroupHasNullMembers` in `TranslatorTest` will fail if we dont check loggroup name is null here, but it does make sense that a log group name can never be null